src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {$match: {$or: {^Array: {of: 
+                  {$and: [
+                    {a: {$gt: {^Inc: {start: 10}}}},
+                    {b: {$gt: {^Inc: {start: 10}}}}
+                  ]},
+                  number: {^Parameter: {Name: "NumberOfClauses", Default: -1}}
+                }}}}
+              ]
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Collection: *Collection
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: *DocumentCount
+        Document:
+          a: [0, 1]
+          b: [2, 3]
+          c: 4
+          d: 5
+          object: {}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50Clauses
+      Repeat: 1000
+      ActivePhase: 3
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith100Clauses
+      Repeat: 1000
+      ActivePhase: 4
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith1000Clauses
+      Repeat: 100
+      ActivePhase: 5
+      NumberOfClauses: 1000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith10000Clauses
+      Repeat: 10
+      ActivePhase: 6
+      NumberOfClauses: 10000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50000Clauses
+      Repeat: 5
+      ActivePhase: 7
+      NumberOfClauses: 50000
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone
```

Comment:
given the size and scale of the workload, this could also be a microbenchmarks test.

Reply:
I thought the usual recommendation was to add all new workloads to genny- has that changed (just for my own knowledge)?

Reply:
We would probably still recommend a larger variant by default.

Reply:
Sorry, bit confused- do you mean a larger evergreen variant here? Not sure how that relates to the question of adding a microbenchmark vs. adding a genny workload. Or did I misunderstand your original comment?

Reply:
that should probably be 'We would probably still recommend *the* larger variant by default.' So standalone is fine.

Reply:
Got it, thank you for clarifying!